app/src/util/dom.ts

```diff
@@ -80,6 +91,43 @@ export const shrinkWrap = (element: HTMLElement | null) => {
   range.setStartBefore(start);
   range.setEndAfter(end);
   const { width } = range.getBoundingClientRect();
-  element.style.width = width + 1 + "px";
+  element.style.width = width + "px";
   element.style.boxSizing = "content-box";
 };
+
+/**
+ * is element covering anything "important" (above anything besides a
+ * "background" element)
+ */
+export const isCovering = (
```

Comment:
Util func that checks a grid of points under an element.

Reply:
Interesting solution, and it seems to work well.

I'm not suggesting changing it, but if you're interested you might consider trying the [intersection observer API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API), which seems to have good support from all the major browsers. Basically it allows you to register a target element that gets checked against all the elements under some node in the DOM (by default it's the browser window, i.e. all elements) for intersections and fires a callback when an intersection occurs. Supposedly it's more performant than checking yourself, but since you're just sampling a few points I think your solution is fine.

Reply:
I'm not sure if ResizeObserver can be used in this manner, because although you can specify a root element other than the viewport, it seems that the root must be an ancestor. That is, the element being covered (let's say an svg chart element) would have to be a child of the table of contents. 

Idk if there is a browser API I could use to make this less kludgy. I could do a simple bbox overlap check, but then I'd have to do it against all the elements in the viewport. It might end up being faster than this method though.

Reply:
I switched to the bbox method. I found the elementsFromPoint method to range from < 1ms to 10ms, and the bbox method to always be < 1ms. Since this function is run in a very debounced fashion, this should never really matter, but it could matter if we needed to call it on every scroll event in the future. The bbox method, with the default "important" element selector, should definitely do less work on average than elementsFromPoint.

Bbox method, at least with the default `section > *` selector, will be less "accurate", in the sense that it's based on DOM hierarchy and not actual page layout, which are not always the same. For example:

```html
<section>
  <!-- narrow box in center of screen. gets checked by util func by default, b/c is a direct descendant of section -->
  <div style="width: 100px">
    <!-- image that is "popped out" of regular document flow, positioned outside its parent div, floating on the left side of the screen. not checked by the util func by default, but could still be under the floating table of contents panel -->
    <img style="position: fixed; left: 0"/>
  </div>
</section>
```

This is unlikely though. If it becomes a problem, could change the selector to `section *` (would have to check many more elements, becoming possibly slower than the elementsFromPoint method).